File: chts-backend/src/main/java/project/chts/springboot/model/Psychological.java
package project.chts.springboot.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Column;

@Entity
@Table(name = "psychological") 

public class Psychological {
	
	
	@Id
	private int child_id;
	
	@Column(name = "Difficulty_In_Reading")
	private String difficultyInReading;
	
	@Column(name = "Difficulty_In_Writing")
	private String difficultyInWriting;
	
	@Column(name = "Hyper_Reactive_Behaviour")
	private String hyperReactiveBehaviour;
	
	@Column(name = "Aggresive_Behaviour")
	private String aggresiveBehaviour;
	
	@Column(name = "Passing_Urine_Bed_Classroom")
	private String passingUrineBedClassroom;
	
	@Column(name = "Poor_Mixing_With_Peers")
	private String poorMixingWithPeers;
	
	@Column(name = "Poor_Eye_Contact")
	private String poorEyeContact;
	
	@Column(name = "Scholostic_Backwardness")
	private String scholosticBackwardness;
	
	@Column(name = " Any_Other")
	private String anyOther;
	
	public Psychological() {
		
		
	}
	
	public Psychological(String difficultyInReading, String difficultyInWriting, String hyperReactiveBehaviour,
			String aggresiveBehaviour, String passingUrineBedClassroom, String poorMixingWithPeers,
			String poorEyeContact, String scholosticBackwardness, String anyOther) {
		super();
		this.difficultyInReading = difficultyInReading;
		this.difficultyInWriting = difficultyInWriting;
		this.hyperReactiveBehaviour = hyperReactiveBehaviour;
		this.aggresiveBehaviour = aggresiveBehaviour;
		this.passingUrineBedClassroom = passingUrineBedClassroom;
		this.poorMixingWithPeers = poorMixingWithPeers;
		this.poorEyeContact = poorEyeContact;
		this.scholosticBackwardness = scholosticBackwardness;
		this.anyOther = anyOther;
	}

	

	public int getChild_id() {
		return child_id;
	}

	public void setChild_id(int child_id) {
		this.child_id = child_id;
	}

	public String getDifficultyInReading() {
		return difficultyInReading;
	}

	public void setDifficultyInReading(String difficultyInReading) {
		this.difficultyInReading = difficultyInReading;
	}

	public String getDifficultyInWriting() {
		return difficultyInWriting;
	}

	public void setDifficultyInWriting(String difficultyInWriting) {
		this.difficultyInWriting = difficultyInWriting;
	}

	public String getHyperReactiveBehaviour() {
		return hyperReactiveBehaviour;
	}

	public void setHyperReactiveBehaviour(String hyperReactiveBehaviour) {
		this.hyperReactiveBehaviour = hyperReactiveBehaviour;
	}

	public String getAggresiveBehaviour() {
		return aggresiveBehaviour;
	}

	public void setAggresiveBehaviour(String aggresiveBehaviour) {
		this.aggresiveBehaviour = aggresiveBehaviour;
	}

	public String getPassingUrineBedClassroom() {
		return passingUrineBedClassroom;
	}

	public void setPassingUrineBedClassroom(String passingUrineBedClassroom) {
		this.passingUrineBedClassroom = passingUrineBedClassroom;
	}

	public String getPoorMixingWithPeers() {
		return poorMixingWithPeers;
	}

	public void setPoorMixingWithPeers(String poorMixingWithPeers) {
		this.poorMixingWithPeers = poorMixingWithPeers;
	}

	public String getPoorEyeContact() {
		return poorEyeContact;
	}

	public void setPoorEyeContact(String poorEyeContact) {
		this.poorEyeContact = poorEyeContact;
	}

	public String getScholosticBackwardness() {
		return scholosticBackwardness;
	}

	public void setScholosticBackwardness(String scholosticBackwardness) {
		this.scholosticBackwardness = scholosticBackwardness;
	}

	public String getAnyOther() {
		return anyOther;
	}

	public void setAnyOther(String anyOther) {
		this.anyOther = anyOther;
	}	
	
}

File: project/src/Components/Psychological.js
import './App.css';
import './New.css';
import React, {useState, useEffect} from 'react';
import TemporaryDrawer from './navbar';
import ScreeningNav from './ScreeningNav';
import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import FormLabel from '@mui/material/FormLabel';
import Stack from '@mui/material/Stack';
import TextField from '@mui/material/TextField';
import { useNavigate, useParams } from 'react-router-dom';
import PsychologicalService from '../Services/PsychologicalService';


const Psychological = () => {

  const {child_id} = useParams();    
  const [difficultyInReading,  setDifficultyInReading] = useState('');
  const [difficultyInWriting,  setDifficultyInWriting] = useState('');
  const [hyperReactiveBehaviour,  setHyperReactiveBehaviour] = useState('');
  const [aggresiveBehaviour,   setAggresiveBehaviour] = useState('');
  const [poorMixingWithPeers,  setPoorMixingWithPeers] = useState('');
  const [poorEyeContact, setPoorEyeContact] = useState('');
  const [scholosticBackwardness,  setScholosticBackwardness] = useState('');
  const [anyOther,  setAnyOther] = useState('');

  const navigate = useNavigate();

  useEffect(() => {
    PsychologicalService.getChildByID(child_id).then((response) => {
      setDifficultyInReading(response.data.difficultyInReading)
      setDifficultyInWriting(response.data.difficultyInWriting)
      setHyperReactiveBehaviour(response.data.hyperReactiveBehaviour)
      setAggresiveBehaviour(response.data.aggresiveBehaviour)
      setPoorMixingWithPeers(response.data.poorMixingWithPeers)
      setPoorEyeContact(response.data.poorEyeContact)
      setScholosticBackwardness(response.data.scholosticBackwardness)
      setAnyOther(response.data.anyOther)
    })  
},[])

const saveChildInfo = (e) => {
  e.preventDefault();

  const ChildInfo = {child_id, difficultyInReading, difficultyInWriting, hyperReactiveBehaviour, aggresiveBehaviour, poorEyeContact, poorMixingWithPeers, scholosticBackwardness, anyOther};

  PsychologicalService.UpdateChild(child_id, ChildInfo).then((response) => {
    console.log(response.data);
    navigate(`/Psychological/${child_id}`);
     }).catch(error => {
    console.log(error);
  })   
}

  return (
    <div>
    
    <TemporaryDrawer/>
    <ScreeningNav child_id ={child_id}/>
    
    <div>
    <h4 className='page-screeninghead'>Psychological Screening</h4>
    <Paper className="form-add">
    <Box
       component="form"
         sx={{
        '& > :not(style)': { m: 1, width: '60ch' },
         }}
           noValidate
          autoComplete="off"
          >
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Difficulty In Reading</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="difficultyInReading" value={difficultyInReading} onChange={(e) => setDifficultyInReading(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value= {'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
         
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Difficulty In Writing</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="difficultyInWriting" value={difficultyInWriting} onChange={(e) => setDifficultyInWriting(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
          
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Hyper Reactive Behaviour</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="hyperReactiveBehaviour" value={hyperReactiveBehaviour} onChange={(e) => setHyperReactiveBehaviour(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
         
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Aggresive Behaviour</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="aggresiveBehaviour" value={aggresiveBehaviour} onChange={(e) => setAggresiveBehaviour(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
         
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Poor Mixing With Peers</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="poorMixingWithPeers" value={poorMixingWithPeers} onChange={(e) => setPoorMixingWithPeers(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
         
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Poor Eye Contact</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="poorEyeContact" value={poorEyeContact} onChange={(e) => setPoorEyeContact(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
     
      </Stack>
        </RadioGroup>
      </FormControl>
          
           <FormControl>
      <FormLabel id="demo-row-radio-buttons-group-label">Scholostic Backwardness</FormLabel>
      <RadioGroup
        row
        aria-labelledby="demo-row-radio-buttons-group-label"
        name="scholosticBackwardness" value={scholosticBackwardness} onChange={(e) => setScholosticBackwardness(e.target.value)}>
        <FormControlLabel value={'Yes'} control={<Radio />} label="Yes" />
        <FormControlLabel value={'No'} control={<Radio />} label="No" />
        <Stack
      component="form"
      sx={{
        width: '25ch',
      }}
      spacing={2}
      noValidate
      autoComplete="off"
    >
      
      </Stack>
        </RadioGroup>
      </FormControl>
          </Box>
          </Paper>

          <Paper className="form-add">
        <Box
       component="form"
         sx={{
        '& > :not(style)': { m: 2, width: '130ch' },
         }}
           noValidate
          autoComplete="off">

      <TextField value ={anyOther} name = "anyOther" onChange={(e) => setAnyOther(e.target.value)} label="Any Other" variant="outlined" />
      </Box>
      </Paper>
      <button type="submit" onClick ={(e) => saveChildInfo(e)} className="btn btn-primary save-button">Save</button>   
    </div>
    </div>
  );
}

export default Psychological;